### PR TITLE
Fix preserve_newlines issue

### PR DESCRIPTION
**Pull request by HookyQR**

### CODE REVIEW

**Comment by HookyQR** [comment]
The revert will have to be dropped for this merge to work. (Or manual merge.) This PR based off the original `preserve_newlines` PR.

File: test/data/css/tests.js [hunk 1]
```diff
@@ -230,6 +230,19 @@ exports.test_data = {
...
         ],
     }, {
+        name: "Preserve Newlines and newline_between_rules",
+        description: "",
+        options: [
+            { name: "preserve_newlines", value: "true" },
+            { name: "newline_between_rules", value: "true" },
+        ],
+        tests: [
+            { input: '.div {}.span {}', output: '.div {}\n\n.span {}' },
```
Comment (bitwiseman) [inline test/data/css/tests.js]: Take these tests and move them into the previous section. 
Then change the matrix options above to  the 4 combinations of `preserve_newlines` and `newline_between_rules`.

File: test/data/css/tests.js [hunk 2]
```diff
@@ -230,6 +230,19 @@ exports.test_data = {
             { input: '#bla, #foo{\n\tcolor:black;{{separator_input}}\tfont-size: 12px;\n}', output: '#bla,\n#foo {\n\tcolor: black;{{separator_output}}\tfont-size: 12px;\n}' }
```
Comment (bitwiseman) [inline test/data/css/tests.js]: The output here has `bla,\n#foo`.  We need to preserve newlines here, too.  So need inputs such as `#bla\n,\n\n#foo` - The newline before the comma should go away, but the other two?

File: test/data/css/tests.js [hunk 3]
```diff
@@ -230,6 +230,19 @@ exports.test_data = {
...
+        name: "Preserve Newlines and newline_between_rules",
+        description: "",
+        options: [
+            { name: "preserve_newlines", value: "true" },
+            { name: "newline_between_rules", value: "true" },
+        ],
+        tests: [
+            { input: '.div {}.span {}', output: '.div {}\n\n.span {}' },
+            { unchanged: '.div {}\n\n.span {}' },
+            { unchanged: '.div {\n\ta: 1;\n\n\n\tb: 2;\n}\n\n\n\n.span {\n\ta: 1;\n}' },
```
Comment (bitwiseman) [inline test/data/css/tests.js]: What about `'.div {\n\n\n\ta:`?

**Comment by bitwiseman** [comment]
Thanks for the fast turn around.  Could you do the `git revert  ba63466` as  the first commit of this PR?   Then it will merge. 
Also, we need to see tests of newline chars in a number of other places, as noted.

File: test/data/css/tests.js (hunk 1 above)
Comment (HookyQR) [inline test/data/css/tests.js]: They don't merge up well because of exactly the issue that requires this fix. I'll add ones from above here.

**Comment by HookyQR** [comment]
Rebased and added tests per request.

File: test/data/css/tests.js (hunk 2 above)
Comment (HookyQR) [inline test/data/css/tests.js]: It seems the original code didn't preserve new lines there anyway. It only preserves new lines between, and within rules, not in the selector list.

File: test/data/css/tests.js (hunk 3 above)
Comment (HookyQR) [inline test/data/css/tests.js]: Added

**Comment by HookyQR** [comment]
Closing this. There is another issue with the preserve new line code. It can remove selectors completely. It just so happens to be hackdays at work today. I believe I know what I'll be doing. ☹️ 😉

**Pull request closed without merging**